pkg/handler/data_recorder_pubsub.go

```diff
@@ -0,0 +1,69 @@
+package handler
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/swagger_gen/models"
+	"google.golang.org/api/option"
+
+	"github.com/sirupsen/logrus"
+)
+
+type pubsubRecorder struct {
+	enabled  bool
+	producer *pubsub.Client
+	topic    *pubsub.Topic
+}
+
+// NewPubsubRecorder creates a new Pubsub recorder
+var NewPubsubRecorder = func() DataRecorder {
+	client, err := pubsub.NewClient(
+		context.Background(),
+		config.Config.RecorderPubsubProjectID,
+		option.WithServiceAccountFile(config.Config.RecorderPubsubKeyFile),
```

Comment:
This method is deprecated, it is recommended to use ```WithCredentialsFile``` instead. Just starting to look into this to check their equivalency, but they seem to operate on the same idea.

As this is an option, does setting it to an empty string affect the use of Application Default Credentials? Would want to make sure that is not the case.

Reply:
I'm sorry, you are right! this option is deprecated I just used a chunk of code I am already using for the past 2 years in many go services here... I am pushing a fix now (thanks I'm also fixing my own services)

Regarding the empty string, yes it will work with default creds, matter effect this is exactly how we develop on our machines using our own accounts with proper IAM